engine/src/main/java/org/terasology/engine/world/chunks/localChunkProvider/RelevanceSystem.java

```diff
@@ -144,20 +145,22 @@ private void updateRelevance() {
      * Add entity to relevance system. create region for it. Update distance if region exists already. Create/Load
      * chunks for region.
      *
-     * @param entity entity to add.
-     * @param distance region's distance.
-     * @param listener chunk relevance listener.
+     * @param entity the region will be centered around the LocationComponent of this entity
+     * @param distance the dimensions of the region, in chunks
+     * @param listener notified when relevant chunks become available
+     *
+     * @return the region of chunks deemed relevant
      */
-    public void addRelevanceEntity(EntityRef entity, Vector3ic distance, ChunkRegionListener listener) {
+    public BlockRegionc addRelevanceEntity(EntityRef entity, Vector3ic distance, ChunkRegionListener listener) {
```

Comment:
this function returns a new BlockRegion. do you want to change this to BlockRegion.

Reply:
Oh, I see why you might ask. Not sure. I'll explain my thought process and you can let me know how it lines up with expected usage convention of the joml and block `*c` types:

ChunkRelevanceRegion is an internal class, and while its `getCurrentRegion` method is public, I don't know anything about whether its return value is supposed to immutable, or always a current view, or fully mutable. If it's not immutable, I didn't want to have to try to figure out how to document the state changes of this internal class to external users of MTE.

I figure if we return a new copy, it's much easier to explain. We know none of the internals are going to change anything about this new copy.

Now maybe you're thinking that because we made this new object explicitly for this method call, we might as well let the caller take ownership and do whatever they want with it, right? No need to use the `*c` interface to restrict the caller to a read-only view of it.

On the other hand, there's no reason _to_ give them a mutable object. A person might expect that if they get back an object from RelevanceSystem with state-changing methods on it, that changing the state of it is supposed to actually _change_ something. And it doesn't. That's why I went with `BlockRegionc`.

I think this is the sort of place that will use a Record class as a return value, when we upgrade to Java 17 and get those.

Reply:
ooh, see yea I guess that makes sense. record class java 17? what is this record you speak of?

Reply:
[Records](https://docs.oracle.com/en/java/javase/16/language/records.html "Java Language Updates for Java 16") for immutable data.